.github/workflows/main.yml

```diff
@@ -14,9 +14,6 @@ jobs:
       matrix:
         ruby:
           - 2.7
-          - 2.6
-          - 2.5
-          - jruby
```

Comment:
Why drop jruby? 🤔

Reply:
From the [faraday_middleware docs](https://github.com/lostisland/faraday_middleware#as-of-v0160-faraday-and-faraday_middleware-no-longer-officially-support-jruby-or-rubinius):

> As of v0.16.0, faraday and faraday_middleware no longer officially support JRuby or Rubinius.

Reply:
I see. It's a pretty big decision to make, only to update a dependency we **chose** to use. This gem is used by many projects out there, some which might be using jruby. I agree dropping support for EOL Ruby versions, but dropping support for something perfectly valid like jruby sounds quite opinionated. Are we sure this is a well-thought decision that we can stand by?

Reply:
Well, the thing here is that whether we can assume or not that clients out there use JRuby, our internal used libraries (in this case Faraday) are dropping support for JRuby. Us, per our internal usage, do not need JRuby and to me it makes sense that if our libraries don't support it, we stop supporting it as well.
(Plus I would add a clear message in the changelog about this).

Reply:
+1 to changelog message if we are going ahead with this @enricoribelli, good shout.

As I see it, we have 3 options:
1. Upgrade faraday and drop support for JRuby.
2. Find an alternative to faraday that supports JRuby.
3. Leave everything as-is and just find a way to silence those deprecation warnings (or at least prevent them from reaching DD).

Reply:
Enrico, an open-sourced project stretches way more than what we currently use internally 🙂 

I agree with everything else, and I'm happy to drop support for jruby if we are all aware of the impact and agree that the best way forward for this gem is without it. It's a conversation that had to happen and I'm happy to see some opinions thrown around.

Since the std HTTP library has likely evolved in all this time, I am wondering whether we couldn't rely on it and thus get rid of an external dependency. Not sure why Faraday was introduced in the first place, but it might be a path worth exploring.

Reply:
I'll have a look at the HTTP library and see if it fits our use cases, would definitely be nice if we could drop the dependency altogether 👍🏻

Reply:
> Enrico, an open-sourced project stretches way more than what we currently use internally 🙂

For sure! But if Faraday (another open-source library 😛  ) drops JRuby support for its clients (us) we can also look that way with our clients:)

Reply:
Of course we can, I'm just saying we should be aware that such a decision involves some discussion and alignment.